### PR TITLE
Switch to setuptools

### PR DESCRIPTION
**Pull request by vidartf**
Allows to build to wheel by `python setup.py bdist_wheel`.

### CODE REVIEW

**Comment by francisco-dlp** [comment]
setuptools should be added to the dependencies then.

**Comment by vidartf** [comment]
Does it make sense to add setuptools to the `setup_requires` as well?

**Comment by francisco-dlp** [comment]
Although it is a chicken and egg problem, I think it does. For example, we could specify the minimum version supported.

**Comment by vidartf** [comment]
This should be all, right?

File: setup.py [hunk 1]
```diff
@@ -170,6 +172,9 @@ def __exit__(self, type, value, traceback):
                   'hyperspy.external.astroML',
                   ],
         requires=install_req,
+        setup_requires=[
+            'setuptools'
```
Comment (jakirkham) [inline setup.py]: I don't understand these lines. How can `setuptools` be in `setup_requires`? Basically this field is only read if `setuptools` exists. So, this is either a no-op or a problem waiting to happen. So, why was this added then?
